File: readability.py
text = input('Text: ')
l = 0
w = 0
s = 0

for i in text:
    if i in ['.', '!', '?']:
        s += 1

for i in text:
    if i == (' '):
        w += 1
w += 1

for i in text:
    if ('a' <= i <= 'z') or ('A' <= i <= 'Z'):
        l += 1

ans = round(5.88 * (l / w) - 29.6 * (s / w) - 15.8)

if ans < 1:
    print('Before Grade 1')
elif ans > 16:
    print('Grade 16+')
else:
    print('Grade', ans)

File: readability.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<math.h>

int roundNo(float num)
{
    return num < 0 ? num - 0.5 : num + 0.5;
}

int main(void)
{
    int l=0;
    int w=0;
    int s=0;
    string text = get_string("Text: ");

    for (int i = 0; i < strlen(text) ; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            s += 1;
        }
    }
    for (int i = 0 ; i < strlen(text) ; i++)
    {
        if (text[i] == ' ')
        {
            w += 1;
        }
    }
    w += 1;

    for (int i = 0 ; i < strlen(text) ; i++)
    {
        if ((65 <= (int) text[i] && (int) text[i] <= 90 )|| (97 <=(int) text[i] && (int) text[i] <= 122))
        {
            l += 1;
        }
    }

    int ans = roundNo(5.88 * (l/w) - 29.6 * (s/w) - 15.8);

    if (ans < 1)
    {
        printf("Before Grade 1");
    }
    else if (ans > 16)
    {
        printf("Grade 16+");
    }
    else
    {
        printf("Grade %i",ans);
    }
}